src/main/java/org/wildfly/security/keystore/KeyStoreUtil.java

```diff
@@ -0,0 +1,166 @@
+     *
+     * The provider supplier is used for loading the key stores.
+     *
+     * @param providers provider supplier for loading the keystore (must not be {@code null})
+     * @param providerName if specified only providers with this name will be used
+     * @param is the key store file input stream (must not be {@code null})
+     * @param filename the filename for prioritizing brute force checks using the file extension
+     * @param password password of the key store
+     * @return loaded key store if recognized
+     * @throws IOException
+     */
+    public static KeyStore loadKeyStore(final Supplier<Provider[]> providers, final String providerName, FileInputStream is, String filename, char[] password) throws IOException {
+
+        DataInputStream dis = new ResetableDataFileInputStream(is);
+
+        int firstInt = dis.readInt();
+        dis.reset();
+
+        KeyStore result = null;
+
+        try {
+            if (firstInt == JKS_MAGIC) {
+                result = tryLoadKeystore(providers, providerName, dis, password, JKS);
+            } else if (firstInt == JCEKS_MAGIC) {
+                result = tryLoadKeystore(providers, providerName, dis, password, JCEKS);
+            } else if (firstInt == VERSION_1 || firstInt == VERSION_2) {
+                dis.reset();
+                dis.skip(32);
+                byte firstElementType = dis.readByte();
+                dis.reset();
+                if (firstElementType <= 5) {
+                    result = tryLoadKeystore(providers, providerName, dis, password, BKS, UBER);
+                } else {
+                    result = tryLoadKeystore(providers, providerName, dis, password, UBER, BKS);
+                }
+
+            } else if (firstInt == VERSION_0) {
+                result = tryLoadKeystore(providers, providerName, dis, password, UBER);
+            } else if ((firstInt & 0xff000000) == SEQUENCE) {
+                String[] parts = filename.split("\\.");
+                String extension = parts[parts.length - 1];
+                if (extension.startsWith("b") || extension.startsWith("B")) {
+                    result = tryLoadKeystore(providers, providerName, dis, password, BCFKS, PKCS12);
+                } else {
+                    result = tryLoadKeystore(providers, providerName, dis, password, PKCS12, BCFKS);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return result;
+    }
+
+    private static KeyStore tryLoadKeystore(final Supplier<Provider[]> providers, final String providerName, InputStream is, char[] password, String... types) {
+        for (String type : types) {
+            log.debug("Searching provider for: " + type);
+            Provider provider = findProvider(providers, providerName, KeyStore.class, type);
+            if (provider == null) {
+                log.debug("Provider not found");
+                return null;
```

Comment:
So in case if I have types= {BCFKS, PKCS12} and I dont have fips bouncy castle registered. Method tries BCFKS and return null. And dont try PKCS12?